### ISSUE TITLE
Enforce code formatting 

### ISSUE THREAD

**Issue by minht11** [Borewit/music-metadata#2191]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature description

My proposal would be:
1. Enforce code style in CI. Add script command to format files.
2. Switch to [Biome](https://biomejs.dev/) for formatting only. It is way faster and has identical or better formatting to prettier.
3. Make tabs default instead of spaces. TLDR. Tabs are more accessable and configurable by the user. More reasoning can be found [here](https://github.com/microsoft/vscode/issues/156304).

The first point is the main part, others are in decreasing importance. Would be willing to provide PR to 1 or all points.

---

Prettier is in project dependencies but does not seem to be used.
In vscode formatting does not even work even if forced, because prettier path does not exist: https://github.com/Borewit/music-metadata/blob/779b319d8d6c422c80b7a5c05ac1bcc9d91822a6/.vscode/settings.json#L7
on my machine changing It to makes it work:
```
  "prettier.prettierPath": "node_modules/prettier/index.cjs",
```

**Comment by Borewit** [comment]
I am interested, the eslint with all it's sub dependencies is very distracting, way to complex to configure to my liking.

Can we postpone the space to tabs, to avoid each and every line will changed with this PR?

Are you willing to do the change for the dependencies (those I own) as well? Preferably starting with the dependencies, and then the dependant packages.

**Comment by minht11** [comment]
Yeah, spaces/tabs can wait.
Just to be clear, you would want to migrate from eslint completely? Biome has most of eslint/unicorn rules, but it is missing some of typescript-eslint rules which involve typechecking. Will PR open peek-reedable to start.

**Comment by Borewit** [comment]
I am open to better alternatives.   
I do believe proper code formatting is important, but I dislike the current complexity to set it up and to maintain it. 
Also the source mapping drives me nuts, for some reason I get totally different lines when debugging unit tests.

These are all things which had working perfectly at certain point, but then due to required updates / dependency deprecation, things just became in opinion just to complex and distracting.

**Comment by Borewit** [comment]
Resolved by #2204
